server/task_executor_schema_update_ghost_sync.go

```diff
@@ -160,27 +159,32 @@ func runGhostMigration(ctx context.Context, server *Server, task *api.Task, stat
 		return true, nil, err
 	}
 
-	waitSync := &sync.WaitGroup{}
-	waitSync.Add(1)
+	syncDone := make(chan struct{})
+	syncError := make(chan error)
 
-	go func(waitSync *sync.WaitGroup) {
-		migrationID, schema, err := executeSync(ctx, task, mi, statement, waitSync)
+	go func() {
+		migrationID, schema, err := executeSync(ctx, task, mi, statement, syncDone)
```

Comment:
Should create a new child context with `context.WithCancel(ctx)`, and explicitly close it if received `syncErr`, because we do not want to wait for the cutover anyway.

Reply:
NM, it does not make sense.